operator/controllers/provision.go

```diff
@@ -0,0 +1,246 @@
+					APIVersion: "V1",
+					Kind:       "Pod",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "logging-plumber-log-aggregator",
+					Namespace: flowTest.ObjectMeta.Namespace,
+					Labels: map[string]string{
+						"app.kubernetes.io/name":            "logging-plumber-log-aggregator",
+						"app.kubernetes.io/managed-by":      "rancher-logging-explorer",
+						"app.kubernetes.io/created-by":      "logging-plumber",
+						"loggingplumber.isala.me/component": "log-aggregator",
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{{
+						Name:  "log-output",
+						Image: "paynejacob/log-output:latest",
+						Ports: []v1.ContainerPort{{
+							Name:          "http",
+							ContainerPort: 80,
+							Protocol:      "TCP",
+						}},
+					}},
+				},
+			}
+			if err := r.Create(ctx, &outputPod); err != nil {
+				logger.Error(err, "failed to create the log output pod")
+				return err
+			}
+			logger.V(1).Info("deployed log output pod", "pod-uuid", outputPod.UID)
+		}
+	} else {
+		logger.V(1).Info("found a already deployed log output pod", "pod-uuid", outputPod.UID)
+	}
+
+	if err := r.deploySlicedFlows(ctx, extraLabels); err != nil {
+		return err
+	}
+
+	flowTest.Status.Status = loggingplumberv1alpha1.Running
+
+	if err := r.Status().Update(ctx, &flowTest); err != nil {
+		logger.Error(err, "failed to update flowtest status")
+		return err
+	}
+
+	return nil
+}
+
+func (r *FlowTestReconciler) deploySlicedFlows(ctx context.Context, extraLabels map[string]string) error {
+	logger := log.FromContext(ctx)
+	flowTest := ctx.Value("flowTest").(loggingplumberv1alpha1.FlowTest)
+
+	var referenceFlow flowv1beta1.Flow
+	if err := r.Get(ctx, types.NamespacedName{
+		Namespace: flowTest.Spec.ReferenceFlow.Namespace,
+		Name:      flowTest.Spec.ReferenceFlow.Name,
+	}, &referenceFlow); err != nil {
+		return err
+	}
+
+	i := 0
+
+	flowTemplate, outTemplate := banzaiTemplates(referenceFlow, flowTest, extraLabels)
+
+	for x := 1; x <= len(referenceFlow.Spec.Match); x++ {
+		targetFlow := *flowTemplate.DeepCopy()
+		targetOutput := *outTemplate.DeepCopy()
+
+		targetFlow.ObjectMeta.Name = fmt.Sprintf("%s-%d-match", flowTest.ObjectMeta.UID, i)
+		targetFlow.ObjectMeta.Labels["loggingplumber.isala.me/test-id"] = fmt.Sprintf("%d", i)
+		targetFlow.ObjectMeta.Labels["loggingplumber.isala.me/test-type"] = "match"
+
+		targetOutput.ObjectMeta.Name = fmt.Sprintf("%s-%d-match", flowTest.ObjectMeta.UID, i)
+		targetOutput.ObjectMeta.Labels["loggingplumber.isala.me/test-id"] = fmt.Sprintf("%d", i)
+		targetOutput.ObjectMeta.Labels["loggingplumber.isala.me/test-type"] = "match"
+		targetOutput.Spec.HTTPOutput.Endpoint = fmt.Sprintf("%s/%s/", targetOutput.Spec.HTTPOutput.Endpoint, targetFlow.ObjectMeta.Name)
+
+		targetFlow.Spec.LocalOutputRefs = []string{targetOutput.ObjectMeta.Name}
+
+		targetFlow.Spec.Match = append(targetFlow.Spec.Match, referenceFlow.Spec.Match[:x]...)
+
+		if err := r.Create(ctx, &targetFlow); err != nil {
```

Comment:
we should create the output before the flow.  This caused issues in older versions of banzai logging